src/txn/txn_recover.c

```diff
@@ -29,6 +29,10 @@ typedef struct {
     u_int max_fileid;  /* Maximum file ID seen. */
     u_int nfiles;      /* Number of files in the metadata. */
 
+    char **remove_uris;          /* Array of tables that should be removed from the metadata. */
+    size_t remove_uris_allocate; /* Allocated size of remove tables array. */
+    u_int nremove_uris;          /* Number of tables marked for removal in the metadata. */
```

Comment:
Maybe that shouldn't be a part of this PR, but I don't think that should be hard. Can we implement a small, lightweight and straightforward `WT_ARRAY` structure to handle structures like this consistently.